app/app.js

```
@@ -121,15 +121,11 @@ module.exports = (options) => {
     const macroParameters = JSON.stringify(exampleConfig.data, null, '\t');
     const componentDirectory = helperFunctions.componentNameToComponentDirectory(componentName);
 
-    try {
-      res.locals.componentView = env.renderString(
-        `
+    res.locals.componentView = !fs.existsSync(path.join(configPaths.components, componentDirectory, 'macro.njk'))
+      ? ''
+      : env.renderString(`
 {% from '${componentDirectory}/macro.njk' import ${macroName} %}
-{{ ${macroName}(${macroParameters}) }}`,
-      );
-    } catch (err) {
-      res.locals.componentView = null;
-    }
+{{ ${macroName}(${macroParameters}) }}`);
```

if there was an error in the nunjucks template then it would swallow it and render a blank page, it also wouldn't log any error- first we tried logging the error then we realised that the reason it was doing that was because some components don't have macros because they are just being used to test stuff or only have js or something like that - we can account for those we think by just checking for the existence of a macro file and in those cases there won't be any rendered component output. Rendering will just be handled by the layoutTemplate

benefit of this is that when there is an error in the template now, it will be displayed in the browser 🙏  so easier to debug

---

Thanks the explanation @oscarduignan !